core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/workflow/WorkflowResource.scala

```diff
@@ -173,31 +170,30 @@ class WorkflowResource {
   @Consumes(Array(MediaType.APPLICATION_JSON))
   def persistWorkflow(workflow: Workflow, @Auth sessionUser: SessionUser): Workflow = {
     val user = sessionUser.getUser
+    // retrieve current workflow from DB
```

Comment:
can we move this block after we check the user?

Reply:
done